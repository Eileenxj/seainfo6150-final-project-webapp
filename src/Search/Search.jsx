import React, { useState } from "react";
import styles from "./Search.module.css";
import SearchResult from "../SearchResults/SearchResults";


const Search = () => {
  const [submittedForm, setSubmittedForm] = useState();
  const [warningInfo, setWarningInfo] = useState();
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  };
const handleChange = (e) => {
  let value = e.target.value;
  if(!(/[^0-9a-zA-Z,]/g.test(value))) {
    setWarningInfo("One or more, comma separated");
  } else {
    setWarningInfo("");
  }
}
  return (
    <div>
      {
        submittedForm ? (
          // <div>Your form entry was {submittedForm.get("query")} and {submittedForm.get("type")}</div>
          <SearchResult query={submittedForm.get("query")} type={submittedForm.get("type")} number={submittedForm.get("number")}
          cuisine={submittedForm.get("cuisine")} equipment={submittedForm.get("equipment")} diet={submittedForm.get("diet")}
          sort={submittedForm.get("sort")} intolerances={submittedForm.get("intolerances")}
          />
        ) : (
          <div className={styles.container}>
          <form onSubmit={onSubmit} className={styles.form}>
            <h3 className={styles.title}>Search Recipes Form</h3>
            <label htmlFor="query" className={styles.label}>Recipe or Keyword</label>
            <input type="text" name="query" id="queryId" className={styles.input} required/><br/>

            <label htmlFor="type" className={styles.label}>Meal Type</label>
            <input type="text" name="type" id="typeId" className={styles.input} 
              onblur="if(this.value.replace(/^ +| +$/g,'')=='')alert('Meal Type can not be empty!')"
              required/><br/>

            <label htmlFor="number" className={styles.label}>The Number of Expected Results</label>
            <input type="text" name="number" id="numberId" className={styles.input} 
              onblur="if(this.value.replace(/^ +| +$/g,'')=='')alert('Number can not be empty!')"
              required/><br/>

            <label htmlFor="cuisine" className={styles.label}>cuisine(optional)</label>
            <input type="text" name="cuisine" id="cuisineId" className={styles.input} 
            onChange={handleChange} placeholder={warningInfo}/><br/>

            <label htmlFor="equipment" className={styles.label}>equipment(optional)</label>
            <input type="text" name="equipment" id="equipmentId" className={styles.input}
            onChange={handleChange} placeholder={warningInfo}/><br/>

            <label htmlFor="dietDropdown" className={styles.label}>Diet(optional)</label><br/>
            <select name="dietDropdown" id="dietDropdown" className={styles.dropdown}>
              <option value="empty">(empty)</option>
              <option value="Gluten Free">Gluten Free</option>
              <option value="Vegetarian">Vegetarian</option>
              <option value="Pescetarian">Pescetarian</option>
            </select><br/>

            <label htmlFor="sortDropdown" className={styles.label}>Sort By(optional)</label><br/>
            <select name="sortDropdown" id="sortDropdown" className={styles.dropdown}>
              <option value="empty">(empty)</option>
              <option value="calories">calories</option>
              <option value="popularity">popularity</option>
              <option value="healthiness">healthiness</option>
            </select><br/>

            <label className={styles.label}>Select Intolerances(optional)</label><br/>
            
            <label for="dairy"> 
              <input type="checkbox" id="dairy" name="dairy" value="dairy"/>
              Dairy 
              <span ></span>
            </label><br/>
           
            <label for="egg"> 
              <input type="checkbox" id="egg" name="egg" value="egg"/>
              Egg
              <span></span>
            </label><br/>
            
            <label for="peanut"> 
              <input type="checkbox" id="peanut" name="peanut" value="peanut"/>
              Peanut
              <span ></span>
            </label><br/>

            <input type="submit" value="Submit" className={styles.submit}/>
            </form >
            </div>
        )
      }
    </div>
  )
};

export default Search;
