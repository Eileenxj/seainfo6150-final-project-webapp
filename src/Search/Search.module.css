.container {
    max-width: 500px;
    width: 100%;
    margin: 0 auto;
    position: relative;
}
.form {
    background: #F9F9F9;
    padding: 25px;
    margin: 150px 0;
    box-shadow: 0 0 20px 0 rgba(0, 0, 0, 0.2), 0 5px 5px 0 rgba(0, 0, 0, 0.24);
}

.title {
    display: block;
    font-size: 30px;
    font-weight: 300;
    margin-bottom: 10px;
}

.label {
    font: 400 12px/16px "Roboto", Helvetica, Arial, sans-serif;
}

.input {
    width: 100%;
    border: 1px solid #ccc;
    background: #FFF;
    margin: 0 0 5px;
    padding: 10px;
}

.dropdown {
    display: inline-block;
    border: 1px solid#ccc;
    border-radius: 4px;
    width: 100%;
    padding: 10px;
    background-color: #ffffff;
    cursor: pointer;
    white-space: nowrap;
}

.submit {
    cursor: pointer;
    width: 100%;
    border: none;
    background: orange;
    color: #FFF;
    margin: 0 0 5px;
    padding: 10px;
    font-size: 15px;
}