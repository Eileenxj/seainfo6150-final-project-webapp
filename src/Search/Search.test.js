import React from "react";
import { render } from "@testing-library/react";
import Search from "./Search.jsx";

it("renders the Search component correctly", () => {
    const { container } = render(<Search/>);
    expect(container).toMatchSnapshot();
  });