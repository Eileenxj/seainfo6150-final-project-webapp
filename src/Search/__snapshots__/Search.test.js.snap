// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Search component correctly 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <form
        class="form"
      >
        <h3
          class="title"
        >
          Search Recipes Form
        </h3>
        <label
          class="label"
          for="query"
        >
          Recipe or Keyword
        </label>
        <input
          class="input"
          id="queryId"
          name="query"
          required=""
          type="text"
        />
        <br />
        <label
          class="label"
          for="type"
        >
          Meal Type
        </label>
        <input
          class="input"
          id="typeId"
          name="type"
          required=""
          type="text"
        />
        <br />
        <label
          class="label"
          for="number"
        >
          The Number of Expected Results
        </label>
        <input
          class="input"
          id="numberId"
          name="number"
          required=""
          type="text"
        />
        <br />
        <label
          class="label"
          for="cuisine"
        >
          cuisine(optional)
        </label>
        <input
          class="input"
          id="cuisineId"
          name="cuisine"
          type="text"
        />
        <br />
        <label
          class="label"
          for="equipment"
        >
          equipment(optional)
        </label>
        <input
          class="input"
          id="equipmentId"
          name="equipment"
          type="text"
        />
        <br />
        <label
          class="label"
          for="dietDropdown"
        >
          Diet(optional)
        </label>
        <br />
        <select
          class="dropdown"
          id="dietDropdown"
          name="dietDropdown"
        >
          <option
            value="empty"
          >
            (empty)
          </option>
          <option
            value="Gluten Free"
          >
            Gluten Free
          </option>
          <option
            value="Vegetarian"
          >
            Vegetarian
          </option>
          <option
            value="Pescetarian"
          >
            Pescetarian
          </option>
        </select>
        <br />
        <label
          class="label"
          for="sortDropdown"
        >
          Sort By(optional)
        </label>
        <br />
        <select
          class="dropdown"
          id="sortDropdown"
          name="sortDropdown"
        >
          <option
            value="empty"
          >
            (empty)
          </option>
          <option
            value="calories"
          >
            calories
          </option>
          <option
            value="popularity"
          >
            popularity
          </option>
          <option
            value="healthiness"
          >
            healthiness
          </option>
        </select>
        <br />
        <label
          class="label"
        >
          Select Intolerances(optional)
        </label>
        <br />
        <label
          for="dairy"
        >
          <input
            id="dairy"
            name="dairy"
            type="checkbox"
            value="dairy"
          />
          Dairy
          <span />
        </label>
        <br />
        <label
          for="egg"
        >
          <input
            id="egg"
            name="egg"
            type="checkbox"
            value="egg"
          />
          Egg
          <span />
        </label>
        <br />
        <label
          for="peanut"
        >
          <input
            id="peanut"
            name="peanut"
            type="checkbox"
            value="peanut"
          />
          Peanut
          <span />
        </label>
        <br />
        <input
          class="submit"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </div>
</div>
`;
